### PR TITLE
[4716] Item icons in dashboard and site explorer does not match

### PR DESCRIPTION
**Pull request by jvega190**
Fix dashboard issues 
- Items not checking on recentlyPublishedDashlet
- Add save dashboard preferences
- Dates not correct when logged as author
- Update PublishEnvProps interface 
- Update freemarker and groovy messages in Icon Guide

https://github.com/craftercms/craftercms/issues/4716

### CODE REVIEW

**Comment by jvega190** [comment]
@rart created PR as draft since I need some confirmation from Vita

File: ui/app/src/utils/state.ts [hunk 1]
```diff
@@ -139,3 +141,13 @@ export function createWidgetDescriptor(widget: WidgetDescriptor): WidgetDescript
...
+
+export function setStoredDashboardPreferences(value: LookupTable<DashboardPreferences>, user: string, site: string) {
+  return window.localStorage.setItem(`craftercms.${user}.dashboardPreferences.${site}`, JSON.stringify(value));
+}
+
+export function getStoredDashboardPreferences(user: string, site: string): LookupTable<DashboardPreferences> {
+  return JSON.parse(window.localStorage.getItem(`craftercms.${user}.dashboardPreferences.${site}`)) as LookupTable<
+    DashboardPreferences
+  >;
+}
```
Comment (rart) [inline ui/app/src/utils/state.ts]: Shouldn't these be per dashlet?

File: ui/app/src/utils/content.ts [hunk 2]
```diff
@@ -256,14 +256,14 @@ export function parseLegacyItemToDetailedItem(item: LegacyItem | LegacyItem[]):
       sizeInBytes: null
     },
     staging: {
-      lastScheduledDate: item.scheduledDate,
-      lastPublishedDate: item.publishedDate,
+      lastScheduledDate: item.scheduledDate ?? item.eventDate,
+      lastPublishedDate: item.publishedDate ?? item.eventDate,
```
Comment (rart) [inline ui/app/src/utils/content.ts]: - `eventDate` can't be both. it's either `datePublished` or `dateScheduled`.
- These prop names are actually outdated. Please update `PublishEnvProps` interface
  -  lastScheduledDate => dateScheduled
  - lastPublishedDate => datePublished
image

File: ui/app/src/utils/content.ts [hunk 3]
```diff
@@ -256,14 +256,14 @@ export function parseLegacyItemToDetailedItem(item: LegacyItem | LegacyItem[]):
...
       publisher: item.user,
       commitId: null
     },
     live: {
-      lastScheduledDate: item.scheduledDate,
-      lastPublishedDate: item.publishedDate,
+      lastScheduledDate: item.scheduledDate ?? item.eventDate,
+      lastPublishedDate: item.publishedDate ?? item.eventDate,
```
Comment (rart) [inline ui/app/src/utils/content.ts]: Same as above.

File: ui/app/src/models/Dashboard.ts [hunk 4]
```diff
@@ -38,3 +38,8 @@ export interface LegacyDeploymentHistoryResponse {
 }
 
 export type LegacyDeploymentHistoryType = 'page' | 'component' | 'document' | 'all';
+
+export interface DashboardPreferences {
+  filterBy: string;
+  numItems?: number;
+}
```
Comment (rart) [inline ui/app/src/models/Dashboard.ts]: Should probably be in each dashlet. Is it the same preference(s) for all?

**Review by rart — CHANGES_REQUESTED** [review]
Store all prefs of all widgets
- ItemsAwaitingApproval: collapsed, showUnpublished
- ApprovedScheduled: collapsed, filter
- RecentlyPublished: collapsed, numOfItems, filter
- MyRecent: collapsed, numOfItems, filter

File: ui/app/src/utils/state.ts [hunk 5]
```diff
@@ -139,3 +141,13 @@ export function createWidgetDescriptor(widget: WidgetDescriptor): WidgetDescript
     uiKey: uuid()
   };
 }
+
+export function setStoredDashboardPreferences(value: LookupTable<DashboardPreferences>, user: string, site: string) {
+  return window.localStorage.setItem(`craftercms.${user}.dashboardPreferences.${site}`, JSON.stringify(value));
```
Comment (rart) [inline ui/app/src/utils/state.ts]:
```suggestion
  return window.localStorage.setItem(`craftercms.dashboard.${dashletId}.${site}.${user}`, JSON.stringify(value));
```

File: ui/app/src/components/IconGuideDashlet/IconGuideDashlet.tsx [hunk 6]
```diff
@@ -51,8 +51,8 @@ const messages = defineMessages({
   component: { id: 'iconGuide.component', defaultMessage: 'Component' },
   folder: { id: 'iconGuide.folder', defaultMessage: 'Folder' },
   levelDescriptor: { id: 'iconGuide.levelDescriptor', defaultMessage: 'levelDescriptor' },
```
Comment (rart) [inline ui/app/src/components/IconGuideDashlet/IconGuideDashlet.tsx]:
```suggestion
  levelDescriptor: { id: 'iconGuide.levelDescriptor', defaultMessage: 'Section Defaults' },
```

**Review by rart — APPROVED** [review]

**Pull request merged**